platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/DiagnosticToProblemListener.java

```diff
@@ -65,9 +68,45 @@ public DiagnosticToProblemListener(InternalProblemReporter problemReporter, Cont
 
     @Override
     public void report(Diagnostic<? extends JavaFileObject> diagnostic) {
+        switch (diagnostic.getKind()) {
+            case ERROR:
+                errorCount++;
+                break;
+            case WARNING:
+            case MANDATORY_WARNING:
+                warningCount++;
+                break;
+            default:
+                break;
+        }
+
         problemReporter.reporting(spec -> buildProblem(diagnostic, spec));
     }
 
+    /**
+     * This method is based on {JavaCompiler#printCount()}
+     */
```

Comment:
❌ The javadoc should state the following:
- functionality: this method prints the number of warnings / errors to the standard output
- intent: when there's no diagnostic listener attached to the compiler object, the number of warnings/errors are part of the putput. Since we have a listener attached, the output changed. Calling this method makes restores compatiblity with the original output
- source: the method implementation is based on {JavaCompiler#printCount()}.

Reply:
Added further documentation